Sources/DDGSync/DDGSyncing.swift

```diff
@@ -179,6 +184,10 @@ public protocol Scheduling {
     func notifyAppLifecycleEvent()
     /// This should be called from externally scheduled background jobs that trigger sync periodically.
     func requestSyncImmediately()
+    /// This should be called when sync needs to be cancelled, e.g. in response to app going to background.
+    func cancelSyncAndSuspendSyncQueue()
+    /// This should be called when sync can be resumed, e.g. in response to app going to foreground.
+    func resumeSyncQueue()
```

Comment:
Note that this API is present in `Scheduling` protocol, i.e. clients would cancel and resume sync via scheduler (e.g. `syncService.scheduler.cancelSyncAndSuspendSyncQueue()`). We could move this API to DDGSyncing, but I thought here would be better as it groups all sync-operation-related APIs in one place. Then again, cancel/resume calls are not "scheduled", so I'd be fine with moving this API to DDGSyncing.